WooCommerce/WooCommerceTests/ViewRelated/MainTabBarControllerTests.swift

```diff
@@ -95,10 +95,72 @@ final class MainTabBarControllerTests: XCTestCase {
         XCTAssertEqual(selectedTabIndexBeforeSiteChange, WooTab.products.visibleIndex())
         XCTAssertEqual(selectedTabIndexAfterSiteChange, WooTab.myStore.visibleIndex())
     }
+
+    func test_when_receiving_a_review_notification_from_a_different_site_navigates_to_reviews_tab_and_presents_review_details() throws {
+        // Arrange
+        let pushNotificationsManager = MockPushNotificationsManager()
+        ServiceLocator.setPushNotesManager(pushNotificationsManager)
+
+        let storesManager = MockStoresManager(sessionManager: .testingInstance)
+        // Reset `receivedActions`
+        storesManager.reset()
+        ServiceLocator.setStores(storesManager)
+
+        // Trigger `viewDidLoad`
+        XCTAssertNotNil(tabBarController.view)
+        stores.updateDefaultStore(storeID: 134)
+
+        // Simulate successful state resetting after logging out from push notification store switching
+        storesManager.whenReceivingAction(ofType: StatsActionV4.self) { action in
+            if case let .resetStoredStats(completion) = action {
+                completion()
+            }
+        }
+        storesManager.whenReceivingAction(ofType: OrderAction.self) { action in
+            if case let .resetStoredOrders(completion) = action {
+                completion()
+            }
+        }
+        storesManager.whenReceivingAction(ofType: ProductReviewAction.self) { action in
+            if case let .resetStoredProductReviews(completion) = action {
+                completion()
+            }
+        }
+
```

Comment:
I'd also suggest a precondition check. But maybe it's already implied. 🤷 

```suggestion

        let reviewsTabNavigationController = tabBarController.tabNavigationController(tab: .reviews)
        assertThat(reviewsTabNavigationController.topViewController, isAnInstanceOf: ReviewsViewController.self)
```

Reply:
Makes sense! Added in: https://github.com/woocommerce/woocommerce-ios/pull/5044/commits/8b9dc633fb1eddabfabb4f251fd3de9aa69849e5